src/Model/Attach.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+/**
+ * @file src/Model/Attach.php
+ * @brief This file contains the Attach class for database interface
+ */
+namespace Friendica\Model;
+
+use Friendica\BaseObject;
+use Friendica\Core\System;
+use Friendica\Core\StorageManager;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Util\Security;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Mimetype;
+
+/**
+ * Class to handle attach dabatase table
+ */
+class Attach extends BaseObject
+{
+
+	/**
+	 * @brief Return a list of fields that are associated with the attach table
+	 *
+	 * @return array field list
+	 */
+	private static function getFields()
+	{
+		$allfields = DBStructure::definition(false);
+		$fields = array_keys($allfields['attach']['fields']);
+		array_splice($fields, array_search('data', $fields), 1);
+		return $fields;
+	}
+
+	/**
+	 * @brief Select rows from the attach table
+	 *
+	 * @param array  $fields     Array of selected fields, empty for all
+	 * @param array  $conditions Array of fields for conditions
+	 * @param array  $params     Array of several parameters
+	 *
+	 * @return boolean|array
+	 *
+	 * @see \Friendica\Database\DBA::select
+	 */
+	public static function select(array $fields = [], array $conditions = [], array $params = [])
+	{
+		if (empty($fields)) {
+			$selected = self::getFields();
+		}
+
+		$r = DBA::select('attach', $fields, $conditions, $params);
+		return DBA::toArray($r);
+	}
+
+	/**
+	 * @brief Retrieve a single record from the attach table
+	 *
+	 * @param array  $fields     Array of selected fields, empty for all
+	 * @param array  $conditions Array of fields for conditions
+	 * @param array  $params     Array of several parameters
+	 *
+	 * @return bool|array
+	 *
+	 * @see \Friendica\Database\DBA::select
+	 */
+	public static function selectFirst(array $fields = [], array $conditions = [], array $params = [])
+	{
+		if (empty($fields)) {
+			$fields = self::getFields();
```

Comment:
`DBA::select` already has a provision for empty field list which means "all the fields in the queried table". You don't need to extract the `attach` table fields and explicitly pass them, you can directly pass the `$fields` parameter to `DBA::selectFirst`.

Reply:
yup, but `self::getFields()` excludes "data" column. Attach model (like Photo model) doesn't return file data while you are looking for metadata. You have to explicitly call `Attach::getData(..)` which, if it's the case, interact with storage backend.

Reply:
Oooooh, I missed that part. Very well then.